### PR TITLE
update nixpkgs (stable: 21.11 -> 22.05)

### PR DESCRIPTION
**Pull request by jonasnick**
Draft because I disabled joinmarket. It causes a couple of errors after the nixpkgs update (assertion error in a "klein" test?). Perhaps the joinmarket package maintainers want to have a look (@nixbitcoin).

### CODE REVIEW

**Comment by erikarvstedt** [comment]
[Fixed branch](https://github.com/erikarvstedt/nix-bitcoin/commits/update-nixpkgs34-ea).

**Review by jonasnick — COMMENTED** [review]
This looks good, thanks a lot @erikarvstedt!

File: flake.nix [hunk 1]
```diff
@@ -67,16 +67,23 @@
...
-        packages = flake-utils.lib.flattenTree (removeAttrs nbPkgs [
-          "pinned" "modulesPkgs" "nixops19_09" "krops" "generate-secrets" "netns-exec"
-        ]) // {
+
+        flakePkgs = (removeAttrs nbPkgs (
+          [
+            "pinned" "modulesPkgs" "nixops19_09" "krops" "generate-secrets" "netns-exec"
+          ] ++ nixpkgs.lib.optionals (system == "aarch64-linux") [
+            # Dependency `klein` is broken on `aarch64-linux`
+            "joinmarket"
```
[the remark this answers is not in the document]
Comment (erikarvstedt) [reply flake.nix:76]: It turns out that klein was broken on all platforms for the same reason (wrong version of `werkzeug`), which I had already fixed.
[Fixed branch](https://github.com/erikarvstedt/nix-bitcoin/commits/update-nixpkgs34-ea), which restores joinmarket on aarch64.

[the remark this answers is not in the document]
Comment (jonasnick) [reply flake.nix]: Excellent. Force pushed and marked ready for review.

**Comment by erikarvstedt** [comment]
[Final fixed branch](https://github.com/erikarvstedt/nix-bitcoin/commits/update-nixpkgs34-ea-2).
This pulls in the latest nixos-22.05 so we don't have to ship a custom `extra-container` derivation.

**Comment by jonasnick** [comment]
Good. Removed leftover `builtins.trace` and pushed the branch.

**Comment by erikarvstedt** [comment]
[Final fixup™](https://github.com/erikarvstedt/nix-bitcoin/commits/update-nixpkgs34-ea-3).

**Pull request merged**